test/functional/feature_discount_ct.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT/X11 software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from decimal import Decimal
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+)
+
+class CTTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 3
+        self.setup_clean_chain = True
+        args = [
+            "-anyonecanspendaremine=1",
+            "-con_blocksubsidy=0",
+            "-con_connect_genesis_outputs=1",
+            "-initialfreecoins=2100000000000000",
+            "-txindex=1",
+        ]
+        self.extra_args = [
+            # node 0 does not accept nor create discounted CTs
+            args + ["-acceptdiscountct=0", "-creatediscountct=0"],
+            # node 1 accepts but does not create discounted CTs
+            args + ["-acceptdiscountct=1", "-creatediscountct=0"],
+            # node 2 both accepts and creates discounted CTs
+            args + ["-acceptdiscountct=1", "-creatediscountct=1"],
+        ]
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def run_test(self):
+        feerate = 1.0
+
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+        node2 = self.nodes[2]
+
+        self.generate(node0, 101)
+        balance = node0.getbalance()
+        assert_equal(balance['bitcoin'], 21000000)
+
+        self.log.info("Create UTXOs")
+        many = {}
+        num = 25
+        for i in range(num):
+            addr = node0.getnewaddress()
+            info = node0.getaddressinfo(addr)
+            many[info['unconfidential']] = 1
+        for i in range(10):
+            addr = node1.getnewaddress()
+            info = node1.getaddressinfo(addr)
+            many[info['unconfidential']] = 1
+        for i in range(10):
+            addr = node2.getnewaddress()
+            info = node2.getaddressinfo(addr)
+            many[info['unconfidential']] = 1
+
+        txid = node0.sendmany("", many)
+        self.generate(node0, 1)
+
+        self.log.info("Send explicit tx to node 0")
+        addr = node0.getnewaddress()
+        info = node0.getaddressinfo(addr)
+        txid = node0.sendtoaddress(info['unconfidential'], 1.0, "", "", False, None, None, None, None, None, None, feerate)
+        tx = node0.gettransaction(txid, True, True)
+        decoded = tx['decoded']
+        vin = decoded['vin']
+        vout = decoded['vout']
+        assert_equal(len(vin), 2)
+        assert_equal(len(vout), 3)
+        assert_equal(tx['fee']['bitcoin'], Decimal('-0.00000326'))
+        assert_equal(decoded['vsize'], 326)
```

Comment:
In 8bafd0c9fa448c3790ed2992f02b8e1245e6c56a:

Everywhere you assert the value of `vsize` can you also assert the value of `discountvsize`?

Reply:
Also in the ordering test in 41abd40b5cdbc910cabac009e4e27e670b2e502a

Reply:
done